packages/math/polygon.ts

```diff
@@ -0,0 +1,227 @@
+import { lineSegment, pointOnLineSegment } from "./segment";
+import type { GlobalPoint, LocalPoint, Polygon } from "./types";
+import { closePolygon, PRECISION } from "./utils";
+
+export function polygon<Point extends GlobalPoint | LocalPoint>(
+  ...points: Point[]
+) {
+  return points as Polygon<Point>;
+}
+
+export function polygonFromPoints<Point extends GlobalPoint | LocalPoint>(
+  points: Point[],
+) {
+  return points as Polygon<Point>;
+}
+
+export function polygonIsClosed<Point extends LocalPoint | GlobalPoint>(
+  polygon: Polygon<Point>,
+) {
+  const first = polygon[0];
+  const last = polygon[polygon.length - 1];
+  return first[0] === last[0] && first[1] === last[1];
+}
+
+export const pointInPolygon = <Point extends LocalPoint | GlobalPoint>(
+  point: Point,
+  polygon: Polygon<Point>,
+) => {
+  const x = point[0];
+  const y = point[1];
+  let inside = false;
+
+  for (let i = 0, j = polygon.length - 1; i < polygon.length; j = i++) {
+    const xi = polygon[i][0];
+    const yi = polygon[i][1];
+    const xj = polygon[j][0];
+    const yj = polygon[j][1];
+
+    if (
+      ((yi > y && yj <= y) || (yi <= y && yj > y)) &&
+      x < ((xj - xi) * (y - yi)) / (yj - yi) + xi
+    ) {
+      inside = !inside;
+    }
+  }
+
+  return inside;
+};
+
+export const pointOnPolygon = <Point extends LocalPoint | GlobalPoint>(
+  p: Point,
+  polygon: Polygon<Point>,
+  threshold = PRECISION,
+) => {
+  let on = false;
+  const closed = closePolygon(polygon);
+
+  for (let i = 0, l = closed.length - 1; i < l; i++) {
+    if (
+      pointOnLineSegment(p, lineSegment(closed[i], closed[i + 1]), threshold)
+    ) {
+      on = true;
+      break;
+    }
+  }
+
+  return on;
+};
+
+// export const polygonInPolygon = <Point extends LocalPoint | GlobalPoint>(
```

Comment:
what's the reason for commenting out these functions?

Reply:
They are unused and I was unsure whether we want to keep these, but after talking to David, I removed them.